---
title: "Saleh_2022_processing"
author: "Alex Brown"
date: "2025-07-22"
output: html_document
---



# Load packages

```{r, include=FALSE}
library(Seurat)
library(SeuratWrappers)
library(tidyverse)
library(Biobase)
library(BiocGenerics)
library(ggplot2)
library(cowplot)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(ggbeeswarm)
library(pheatmap)
library(viridis)
library(EnhancedVolcano)
library(scDblFinder)
library(UCell)
```



# Read in the data (From Saleh et al. 2022; doi: 10.1016/j.usci.2022.105415)

```{r, include=FALSE}
wt_1_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/WT_1_filtered_feature_bc_matrix")

wt_2_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/WT_2_filtered_feature_bc_matrix")

mdx_1_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/mdx_1_filtered_feature_bc_matrix")

mdx_2_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/mdx_2_filtered_feature_bc_matrix")

mdxD2_1_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/mdxD2_1_filtered_feature_bc_matrix")

mdxD2_2_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Saleh_2022_scRNAseq/data/mdxD2_2_filtered_feature_bc_matrix")
```



# Create Seurat Object

```{r, include=FALSE}
wt_1_seurat <- CreateSeuratObject(counts = wt_1_data, min.cells = 3, project = "WT_1")

wt_2_seurat <- CreateSeuratObject(counts = wt_2_data, min.cells = 3, project = "WT_2")

mdx_1_seurat <- CreateSeuratObject(counts = mdx_1_data, min.cells = 3, project = "mdx_1")

mdx_2_seurat <- CreateSeuratObject(counts = mdx_2_data, min.cells = 3, project = "mdx_2")

mdxD2_1_seurat <- CreateSeuratObject(counts = mdxD2_1_data, min.cells = 3, project = "mdxD2_1")

mdxD2_2_seurat <- CreateSeuratObject(counts = mdxD2_2_data, min.cells = 3, project = "mdxD2_2")
```



# 1. QC

## % mitochondrial reads

```{r, include=FALSE}
wt_1_seurat[["percent.mito"]] <- PercentageFeatureSet(wt_1_seurat, pattern = "^mt-")
wt_2_seurat[["percent.mito"]] <- PercentageFeatureSet(wt_2_seurat, pattern = "^mt-")
mdx_1_seurat[["percent.mito"]] <- PercentageFeatureSet(mdx_1_seurat, pattern = "^mt-")
mdx_2_seurat[["percent.mito"]] <- PercentageFeatureSet(mdx_2_seurat, pattern = "^mt-")
mdxD2_1_seurat[["percent.mito"]] <- PercentageFeatureSet(mdxD2_1_seurat, pattern = "^mt-")
mdxD2_2_seurat[["percent.mito"]] <- PercentageFeatureSet(mdxD2_2_seurat, pattern = "^mt-")
```



# 2. Filtering

```{r, include=FALSE}
wt_1_seurat_filtered <- subset(wt_1_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
wt_2_seurat_filtered <- subset(wt_2_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
mdx_1_seurat_filtered <- subset(mdx_1_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
mdx_2_seurat_filtered <- subset(mdx_2_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
mdxD2_1_seurat_filtered <- subset(mdxD2_1_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
mdxD2_2_seurat_filtered <- subset(mdxD2_2_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
```



# 3. Pre-process

```{r, include=FALSE}
all_genes_wt_1 <- rownames(wt_1_seurat_filtered)
all_genes_wt_2 <- rownames(wt_2_seurat_filtered)
all_genes_mdx_1 <- rownames(mdx_1_seurat_filtered)
all_genes_mdx_2 <- rownames(mdx_2_seurat_filtered)
all_genes_mdxD2_1 <- rownames(mdxD2_1_seurat_filtered)
all_genes_mdxD2_2 <- rownames(mdxD2_2_seurat_filtered)
```


## WT_1

```{r, include=FALSE}
wt_1_seurat_norm <- wt_1_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_wt_1) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(wt_1_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
wt_1_seurat_norm <- wt_1_seurat_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## WT_2

```{r, include=FALSE}
wt_2_seurat_norm <- wt_2_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_wt_2) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(wt_2_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
wt_2_seurat_norm <- wt_2_seurat_norm %>%
  FindNeighbors(dims = 1:41) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:41)
```


## mdx_1

```{r, include=FALSE}
mdx_1_seurat_norm <- mdx_1_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_mdx_1) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(mdx_1_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
mdx_1_seurat_norm <- mdx_1_seurat_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## mdx_2

```{r, include=FALSE}
mdx_2_seurat_norm <- mdx_2_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_mdx_2) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(mdx_2_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
mdx_2_seurat_norm <- mdx_2_seurat_norm %>%
  FindNeighbors(dims = 1:36) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:36)
```


## mdxD2_1

```{r, include=FALSE}
mdxD2_1_seurat_norm <- mdxD2_1_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_mdxD2_1) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(mdxD2_1_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
mdxD2_1_seurat_norm <- mdxD2_1_seurat_norm %>%
  FindNeighbors(dims = 1:40) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:40)
```


## mdxD2_2

```{r, include=FALSE}
mdxD2_2_seurat_norm <- mdxD2_2_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_mdxD2_2) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(mdxD2_2_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
mdxD2_2_seurat_norm <- mdxD2_2_seurat_norm %>%
  FindNeighbors(dims = 1:39) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:39)
```



# 4. scDblFinder

```{r, include=FALSE}
wt_1_dbl <- scDblFinder(GetAssayData(wt_1_seurat_norm, slot = "counts"), clusters = Idents(wt_1_seurat_norm))

wt_1_seurat_norm$scDblFinder.class <- wt_1_dbl$scDblFinder.class


wt_2_dbl <- scDblFinder(GetAssayData(wt_2_seurat_norm, slot = "counts"), clusters = Idents(wt_2_seurat_norm))

wt_2_seurat_norm$scDblFinder.class <- wt_2_dbl$scDblFinder.class


mdx_1_dbl <- scDblFinder(GetAssayData(mdx_1_seurat_norm, slot = "counts"), clusters = Idents(mdx_1_seurat_norm))

mdx_1_seurat_norm$scDblFinder.class <- mdx_1_dbl$scDblFinder.class


mdx_2_dbl <- scDblFinder(GetAssayData(mdx_2_seurat_norm, slot = "counts"), clusters = Idents(mdx_2_seurat_norm))

mdx_2_seurat_norm$scDblFinder.class <- mdx_2_dbl$scDblFinder.class


mdxD2_1_dbl <- scDblFinder(GetAssayData(mdxD2_1_seurat_norm, slot = "counts"), clusters = Idents(mdxD2_1_seurat_norm))

mdxD2_1_seurat_norm$scDblFinder.class <- mdxD2_1_dbl$scDblFinder.class


mdxD2_2_dbl <- scDblFinder(GetAssayData(mdxD2_2_seurat_norm, slot = "counts"), clusters = Idents(mdxD2_2_seurat_norm))

mdxD2_2_seurat_norm$scDblFinder.class <- mdxD2_2_dbl$scDblFinder.class
```


## Remove Doublets

```{r, include=FALSE}
wt_1_singlets <- subset(wt_1_seurat_norm, subset = scDblFinder.class == "singlet")

wt_2_singlets <- subset(wt_2_seurat_norm, subset = scDblFinder.class == "singlet")

mdx_1_singlets <- subset(mdx_1_seurat_norm, subset = scDblFinder.class == "singlet")

mdx_2_singlets <- subset(mdx_2_seurat_norm, subset = scDblFinder.class == "singlet")

mdxD2_1_singlets <- subset(mdxD2_1_seurat_norm, subset = scDblFinder.class == "singlet")

mdxD2_2_singlets <- subset(mdxD2_2_seurat_norm, subset = scDblFinder.class == "singlet")
```



# 5. Combine conditions

## Merge data

```{r, include=FALSE}
ordered_datasets <- c(wt_1_singlets, wt_2_singlets, mdx_1_singlets, mdx_2_singlets, mdxD2_1_singlets, mdxD2_2_singlets)

features_merge <- SelectIntegrationFeatures(object.list = ordered_datasets)
```


## Find anchors

```{r, include=FALSE}
anchors_merge <- FindIntegrationAnchors(object.list = list(wt_1_singlets, wt_2_singlets, mdx_1_singlets, mdx_2_singlets, mdxD2_1_singlets, mdxD2_2_singlets), anchor.features = features_merge)
```


## Combine data

```{r, include=FALSE}
mdx_combined <- IntegrateData(anchorset = anchors_merge)

DefaultAssay(mdx_combined) <- "integrated"
```



# 6. Reprocess

## Scale data and run PCA
```{r, include=FALSE}
all_genes <- rownames(mdx_combined)
mdx_combined <- ScaleData(mdx_combined, features = all_genes)
mdx_combined <- RunPCA(mdx_combined, npcs = 50)
```

```{r}
ElbowPlot(mdx_combined, ndims = 50) ### to check dims
```


## Cluster

```{r, include=FALSE}
mdx_combined <- mdx_combined %>%
  FindNeighbors(dims = 1:37) %>%
  FindClusters(resolution = 0.1)

Idents(mdx_combined) <- "seurat_clusters"

umap.method = 'umap-learn'
metric = 'correlation'

mdx_combined <- RunUMAP(mdx_combined, reduction = "pca", dims = 1:37)
```



# 7. Annotate cells

```{r, include=FALSE}
cluster_ids <- levels(mdx_combined$seurat_clusters)
```


## Order conditions

```{r, include=FALSE}
mdx_combined$orig.ident <- factor(mdx_combined$orig.ident, levels = c("WT_1", "WT_2", "mdx_1", "mdx_2", "mdxD2_1", "mdxD2_2"))
```


## Combine conditions

```{r, include=FALSE}
DefaultAssay(mdx_combined) <- "RNA"

Idents(mdx_combined) <- mdx_combined$seurat_clusters

mdx_combined <- JoinLayers(mdx_combined, what = "all")


mdx_combined$orig.ident <- ifelse(
  mdx_combined$orig.ident %in% c("WT_1", "WT_2"), "WT",
  ifelse(
    mdx_combined$orig.ident %in% c("mdx_1", "mdx_2"), "mdx",
    ifelse(
      mdx_combined$orig.ident %in% c("mdxD2_1", "mdxD2_2"), "mdxD2",
      mdx_combined$orig.ident)))

mdx_combined$orig.ident <- factor(mdx_combined$orig.ident, levels = c("WT", "mdx", "mdxD2"))
```


## Annotate clusters

```{r}
DefaultAssay(mdx_combined) <- "RNA"
Idents(mdx_combined) <- "seurat_clusters"

DotPlot(object = mdx_combined, features = c("Pax7", "Acta1", "Cdh5", "Myl9", "Pdgfrb", "Pdgfra", "Cma1", "Cd68", "Csf1r", "S100a9", "Il7r", "Ptn", "Gypa")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, face = "italic")) +
  scale_color_gradientn(colors = c("red", "white", "blue")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, face = "italic"))
```


```{r, include=FALSE}
cluster_labels <- c("Monocytes/macrophages", 
                 "FAPs", 
                 "Endothelial cells", 
                 "Monocytes/macrophages",
                 "Myonuclei",
                 "Erythrocytes",
                 "Neural/glial cells",
                 "Neutrophils",
                 "MuSCs",
                 "NK cells",
                 "vSMCs/pericytes",
                 "FAPs",
                 "Mast cells")

cluster_to_cell_type <- setNames(cluster_labels, cluster_ids)

mdx_combined$seurat_clusters <- cluster_labels[mdx_combined$seurat_clusters]

mdx_combined$seurat_clusters <- factor(mdx_combined$seurat_clusters, levels = c("MuSCs", 
            "Myonuclei", 
            "Endothelial cells",
            "vSMCs/pericytes",
            "FAPs",
            "Mast cells",
            "Monocytes/macrophages",
            "Neutrophils",
            "NK cells",
            "Neural/glial cells",
            "Erythrocytes"))
```



# 8. Subset MuSCs

```{r, include=FALSE}
Idents(mdx_combined) <- "seurat_clusters"

mdx_MuSCs <- subset(mdx_combined, idents = "MuSCs")
```


## DEGs per condition

```{r, include=FALSE}
Idents(mdx_MuSCs) <- "orig.ident"

DEGs_mdx_MuSCs <- FindAllMarkers(object = mdx_MuSCs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```



# Cell scoring

## Create gene list
```{r, include=FALSE}
cachexia_gene_list <- cachexia_gene_list <- read.csv("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/MuSCs_combined/DEGs_MuSCs_combined_subclusters.csv")

cachexia_gene_list <- cachexia_gene_list[cachexia_gene_list$cluster == "Cachectic" & cachexia_gene_list$p_val_adj < 0.05, ]

cachexia_gene_list <- cachexia_gene_list$gene
```


## AddModuleScore by orig.ident

```{r, include=FALSE}
cachexia_score <- AddModuleScore_UCell(mdx_MuSCs, features = list(cachexia_gene_list), name = "cachexia_score")

means_cachexia_score <- cachexia_score@meta.data %>%
  group_by(orig.ident) %>%
  summarise(average_cachexia_score = mean(signature_1cachexia_score, na.rm = TRUE))
```

```{r}
print(means_cachexia_score)
```


```{r}
VlnPlot(cachexia_score, features = "signature_1cachexia_score", group.by = "orig.ident") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45)) +
  labs(x = NULL, y = "Cachexia Score", title = NULL) +
  geom_hline(yintercept = 0, linetype = "solid", color = "black")
```



# Visualize cluster proportions to see if it matches the paper

```{r, include=FALSE}
mdx_cluster_proportions <- mdx_combined@meta.data %>%
  dplyr::select(orig.ident, seurat_clusters) %>%
  group_by(orig.ident, seurat_clusters) %>%
  summarize(count = n()) %>%
  group_by(orig.ident) %>%
  mutate(percentage = count / sum(count) * 100)
```

```{r}
mdx_cluster_proportions$orig.ident <- factor(
  mdx_cluster_proportions$orig.ident,
  levels = c("WT", "mdx", "mdxD2"),
  labels = c("WT", "mdx", "mdxD2"))

ggplot(mdx_cluster_proportions, aes(x = orig.ident, y = percentage, fill = seurat_clusters)) +
  geom_bar(stat = "identity", width = 0.75, color = "black") +
  scale_y_continuous(labels = scales::percent_format(scale = 1), expand = c(0, 0)) +
  labs(title = "", x = "", y = "Proportion") +
  theme(legend.position = "right",  panel.background = element_blank(), axis.line = element_line(color = "black"), axis.text.x = element_text(angle = 45, hjust = 1)) +
  guides(fill = guide_legend(title = NULL))
```

### It does!