---
title: "Walter_2024_scRNAseq"
author: "Alex Brown"
date: "2025-07-24"
output: html_document
---



# Load packages

```{r, include=FALSE}
library(Seurat)
library(SeuratWrappers)
library(tidyverse)
library(Biobase)
library(BiocGenerics)
library(ggplot2)
library(cowplot)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(ggbeeswarm)
library(pheatmap)
library(viridis)
library(EnhancedVolcano)
library(scDblFinder)
library(UCell)
library(harmony)
```



# Read in the data (from Walter et al. 2024; doi 10.1038/s43587-024-00756-3)

### Young (3) 0 dpi from GSE143437 (De Micheli et al. 2020; doi 10.1016/j.celrep.2020.02.067)

```{r, include=FALSE}
## Read in raw data and meta data
young_rawdata <- read.delim("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Young/GSE143437_DeMicheli_MuSCatlas_rawdata.txt.gz", row.names = 1)

young_metadata <- read.delim("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Young/GSE143437_DeMicheli_MuSCatlas_metadata.txt.gz", row.names = 1)


## Subset for uninjured
young_d0_metadata <- subset(young_metadata, sampleID %in% c("D0_A", "D0_B", "D0_Cv3"))

d0_cells <- rownames(young_d0_metadata)

young_d0_rawdata <- young_rawdata[, colnames(young_rawdata) %in% d0_cells]

young_d0_metadata <- young_d0_metadata[colnames(young_d0_rawdata), ]


## Remove others with injured muscle
rm(young_rawdata, young_metadata, d0_cells)
```

### Old (4) 0 dpi from GSE162172 (Walter et al. 2024)

```{r, include=FALSE}
Old_A_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Old_D0_A_filtered_feature_bc_matrix")

Old_B_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Old_D0_B_filtered_feature_bc_matrix")

Old_C_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Old_D0_C_filtered_feature_bc_matrix")

Old_D_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Old_D0_D_filtered_feature_bc_matrix")
```


### Geriatric (4) 0 dpi from GSE232106 (Walter et al. 2024)

```{r, include=FALSE}
Ger_A_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Ger_D0_A_filtered_feature_bc_matrix")

Ger_B_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Ger_D0_B_filtered_feature_bc_matrix")

Ger_C_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Ger_D0_C_filtered_feature_bc_matrix")

Ger_D_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/Walter_2024_scRNAseq/data/Ger_D0_D_filtered_feature_bc_matrix")
```



# Create Seurat Object

```{r, include=FALSE}
## Young
### NOTE - already have cell_annotations

Young_seurat <- CreateSeuratObject(counts = young_d0_rawdata, meta.data = young_d0_metadata, project = "MuSC_D0")

Young_seurat_IDs <- SplitObject(Young_seurat, split.by = "sampleID")

Young_A_seurat <- Young_seurat_IDs[["D0_A"]]
Young_B_seurat <- Young_seurat_IDs[["D0_B"]]
Young_C_seurat <- Young_seurat_IDs[["D0_Cv3"]]


## Old

Old_A_seurat <- CreateSeuratObject(counts = Old_A_data, min.cells = 3, project = "Old_A")

Old_B_seurat <- CreateSeuratObject(counts = Old_B_data, min.cells = 3, project = "Old_B")

Old_C_seurat <- CreateSeuratObject(counts = Old_C_data, min.cells = 3, project = "Old_C")

Old_D_seurat <- CreateSeuratObject(counts = Old_D_data, min.cells = 3, project = "Old_D")


## Geriatric

Ger_A_seurat <- CreateSeuratObject(counts = Ger_A_data, min.cells = 3, project = "Ger_A")

Ger_B_seurat <- CreateSeuratObject(counts = Ger_B_data, min.cells = 3, project = "Ger_B")

Ger_C_seurat <- CreateSeuratObject(counts = Ger_C_data, min.cells = 3, project = "Ger_C")

Ger_D_seurat <- CreateSeuratObject(counts = Ger_D_data, min.cells = 3, project = "Ger_D")
```



# 1. QC

## % mitochondrial reads

```{r, include=FALSE}
Young_A_seurat[["percent.mito"]] <- PercentageFeatureSet(Young_A_seurat, pattern = "^mt-")

Young_B_seurat[["percent.mito"]] <- PercentageFeatureSet(Young_B_seurat, pattern = "^mt-")

Young_C_seurat[["percent.mito"]] <- PercentageFeatureSet(Young_C_seurat, pattern = "^mt-")

Old_A_seurat[["percent.mito"]] <- PercentageFeatureSet(Old_A_seurat, pattern = "^mt-")

Old_B_seurat[["percent.mito"]] <- PercentageFeatureSet(Old_B_seurat, pattern = "^mt-")

Old_C_seurat[["percent.mito"]] <- PercentageFeatureSet(Old_C_seurat, pattern = "^mt-")

Old_D_seurat[["percent.mito"]] <- PercentageFeatureSet(Old_D_seurat, pattern = "^mt-")

Ger_A_seurat[["percent.mito"]] <- PercentageFeatureSet(Ger_A_seurat, pattern = "^mt-")

Ger_B_seurat[["percent.mito"]] <- PercentageFeatureSet(Ger_B_seurat, pattern = "^mt-")

Ger_C_seurat[["percent.mito"]] <- PercentageFeatureSet(Ger_C_seurat, pattern = "^mt-")

Ger_D_seurat[["percent.mito"]] <- PercentageFeatureSet(Ger_D_seurat, pattern = "^mt-")
```



# 2. Filtering

### NOTE - percent.mito < 25% (instead of 10% with all others) because it otherwise would filter out way too many cells. 

```{r, include=FALSE}
Young_A_seurat_filtered <- subset(Young_A_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Young_B_seurat_filtered <- subset(Young_B_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Young_C_seurat_filtered <- subset(Young_C_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Old_A_seurat_filtered <- subset(Old_A_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Old_B_seurat_filtered <- subset(Old_B_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Old_C_seurat_filtered <- subset(Old_C_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Old_D_seurat_filtered <- subset(Old_D_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Ger_A_seurat_filtered <- subset(Ger_A_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Ger_B_seurat_filtered <- subset(Ger_B_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Ger_C_seurat_filtered <- subset(Ger_C_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)

Ger_D_seurat_filtered <- subset(Ger_D_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 25)
```



# 3. Pre-process

```{r, include=FALSE}
all_genes_Young_A <- rownames(Young_A_seurat_filtered)
all_genes_Young_B <- rownames(Young_B_seurat_filtered)
all_genes_Young_C <- rownames(Young_C_seurat_filtered)
all_genes_Old_A <- rownames(Old_A_seurat_filtered)
all_genes_Old_B <- rownames(Old_B_seurat_filtered)
all_genes_Old_C <- rownames(Old_C_seurat_filtered)
all_genes_Old_D <- rownames(Old_D_seurat_filtered)
all_genes_Ger_A <- rownames(Ger_A_seurat_filtered)
all_genes_Ger_B <- rownames(Ger_B_seurat_filtered)
all_genes_Ger_C <- rownames(Ger_C_seurat_filtered)
all_genes_Ger_D <- rownames(Ger_D_seurat_filtered)
```


## Young_A

```{r, include=FALSE}
Young_A_seurat_norm <- Young_A_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Young_A) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Young_A_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Young_A_seurat_norm <- Young_A_seurat_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## Young_B

```{r, include=FALSE}
Young_B_seurat_norm <- Young_B_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Young_B) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Young_B_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Young_B_seurat_norm <- Young_B_seurat_norm %>%
  FindNeighbors(dims = 1:37) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:37)
```


## Young_C

```{r, include=FALSE}
Young_C_seurat_norm <- Young_C_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Young_C) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Young_C_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Young_C_seurat_norm <- Young_C_seurat_norm %>%
  FindNeighbors(dims = 1:41) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:41)
```


## Old_A

```{r, include=FALSE}
Old_A_seurat_norm <- Old_A_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Old_A) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Old_A_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Old_A_seurat_norm <- Old_A_seurat_norm %>%
  FindNeighbors(dims = 1:35) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:35)
```


## Old_B

```{r, include=FALSE}
Old_B_seurat_norm <- Old_B_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Old_B) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Old_B_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Old_B_seurat_norm <- Old_B_seurat_norm %>%
  FindNeighbors(dims = 1:40) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:40)
```


## Old_C

```{r, include=FALSE}
Old_C_seurat_norm <- Old_C_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Old_C) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Old_C_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Old_C_seurat_norm <- Old_C_seurat_norm %>%
  FindNeighbors(dims = 1:41) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:41)
```


## Old_D

```{r, include=FALSE}
Old_D_seurat_norm <- Old_D_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Old_D) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Old_D_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Old_D_seurat_norm <- Old_D_seurat_norm %>%
  FindNeighbors(dims = 1:42) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:42)
```


## Ger_A

```{r, include=FALSE}
Ger_A_seurat_norm <- Ger_A_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Ger_A) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Ger_A_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Ger_A_seurat_norm <- Ger_A_seurat_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## Ger_B

```{r, include=FALSE}
Ger_B_seurat_norm <- Ger_B_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Ger_B) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Ger_B_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Ger_B_seurat_norm <- Ger_B_seurat_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## Ger_C

```{r, include=FALSE}
Ger_C_seurat_norm <- Ger_C_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Ger_C) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Ger_C_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Ger_C_seurat_norm <- Ger_C_seurat_norm %>%
  FindNeighbors(dims = 1:42) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:42)
```


## Ger_D

```{r, include=FALSE}
Ger_D_seurat_norm <- Ger_D_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_Ger_D) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(Ger_D_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
Ger_D_seurat_norm <- Ger_D_seurat_norm %>%
  FindNeighbors(dims = 1:33) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:33)
```



# 4. scDblFinder

```{r, include=FALSE}
Young_A_dbl <- scDblFinder(GetAssayData(Young_A_seurat_norm, slot = "counts"), clusters = Idents(Young_A_seurat_norm))

Young_A_seurat_norm$scDblFinder.class <- Young_A_dbl$scDblFinder.class


Young_B_dbl <- scDblFinder(GetAssayData(Young_B_seurat_norm, slot = "counts"), clusters = Idents(Young_B_seurat_norm))

Young_B_seurat_norm$scDblFinder.class <- Young_B_dbl$scDblFinder.class


Young_C_dbl <- scDblFinder(GetAssayData(Young_C_seurat_norm, slot = "counts"), clusters = Idents(Young_C_seurat_norm))

Young_C_seurat_norm$scDblFinder.class <- Young_C_dbl$scDblFinder.class


Old_A_dbl <- scDblFinder(GetAssayData(Old_A_seurat_norm, slot = "counts"), clusters = Idents(Old_A_seurat_norm))

Old_A_seurat_norm$scDblFinder.class <- Old_A_dbl$scDblFinder.class


Old_B_dbl <- scDblFinder(GetAssayData(Old_B_seurat_norm, slot = "counts"), clusters = Idents(Old_B_seurat_norm))

Old_B_seurat_norm$scDblFinder.class <- Old_B_dbl$scDblFinder.class


Old_C_dbl <- scDblFinder(GetAssayData(Old_C_seurat_norm, slot = "counts"), clusters = Idents(Old_C_seurat_norm))

Old_C_seurat_norm$scDblFinder.class <- Old_C_dbl$scDblFinder.class


Old_D_dbl <- scDblFinder(GetAssayData(Old_D_seurat_norm, slot = "counts"), clusters = Idents(Old_D_seurat_norm))

Old_D_seurat_norm$scDblFinder.class <- Old_D_dbl$scDblFinder.class


Ger_A_dbl <- scDblFinder(GetAssayData(Ger_A_seurat_norm, slot = "counts"), clusters = Idents(Ger_A_seurat_norm))

Ger_A_seurat_norm$scDblFinder.class <- Ger_A_dbl$scDblFinder.class


Ger_B_dbl <- scDblFinder(GetAssayData(Ger_B_seurat_norm, slot = "counts"), clusters = Idents(Ger_B_seurat_norm))

Ger_B_seurat_norm$scDblFinder.class <- Ger_B_dbl$scDblFinder.class


Ger_C_dbl <- scDblFinder(GetAssayData(Ger_C_seurat_norm, slot = "counts"), clusters = Idents(Ger_C_seurat_norm))

Ger_C_seurat_norm$scDblFinder.class <- Ger_C_dbl$scDblFinder.class


Ger_D_dbl <- scDblFinder(GetAssayData(Ger_D_seurat_norm, slot = "counts"), clusters = Idents(Ger_D_seurat_norm))

Ger_D_seurat_norm$scDblFinder.class <- Ger_D_dbl$scDblFinder.class
```



## Remove Doublets

```{r, include=FALSE}
Young_A_singlets <- subset(Young_A_seurat_norm, subset = scDblFinder.class == "singlet")

Young_B_singlets <- subset(Young_B_seurat_norm, subset = scDblFinder.class == "singlet")

Young_C_singlets <- subset(Young_C_seurat_norm, subset = scDblFinder.class == "singlet")

Old_A_singlets <- subset(Old_A_seurat_norm, subset = scDblFinder.class == "singlet")

Old_B_singlets <- subset(Old_B_seurat_norm, subset = scDblFinder.class == "singlet")

Old_C_singlets <- subset(Old_C_seurat_norm, subset = scDblFinder.class == "singlet")

Old_D_singlets <- subset(Old_D_seurat_norm, subset = scDblFinder.class == "singlet")

Ger_A_singlets <- subset(Ger_A_seurat_norm, subset = scDblFinder.class == "singlet")

Ger_B_singlets <- subset(Ger_B_seurat_norm, subset = scDblFinder.class == "singlet")

Ger_C_singlets <- subset(Ger_C_seurat_norm, subset = scDblFinder.class == "singlet")

Ger_D_singlets <- subset(Ger_D_seurat_norm, subset = scDblFinder.class == "singlet")
```



# 5. Combine conditions and integrate with Harmony

## Remove unnecessary objects to improve processing

```{r, include=FALSE}
rm(all_genes_Young_A, all_genes_Young_B, all_genes_Young_C, all_genes_Old_A, all_genes_Old_B, all_genes_Old_C, all_genes_Old_D, all_genes_Ger_A, all_genes_Ger_B, all_genes_Ger_C, all_genes_Ger_D, Ger_A_data, Ger_A_dbl, Ger_A_seurat, Ger_A_seurat_filtered, Ger_A_seurat_norm, Ger_B_data, Ger_B_dbl, Ger_B_seurat, Ger_B_seurat_filtered, Ger_B_seurat_norm, Ger_C_data, Ger_C_dbl, Ger_C_seurat, Ger_C_seurat_filtered, Ger_C_seurat_norm, Ger_D_data, Ger_D_dbl, Ger_D_seurat, Ger_D_seurat_filtered, Ger_D_seurat_norm, Young_A_dbl, Young_A_seurat, Young_A_seurat_filtered, Young_A_seurat_norm, Young_B_dbl, Young_B_seurat, Young_B_seurat_filtered, Young_B_seurat_norm, Young_C_seurat, Young_C_seurat_filtered, Young_C_seurat_norm, young_d0_metadata, young_d0_rawdata, Young_seurat, Young_seurat_IDs, Old_A_data, Old_A_dbl, Old_A_seurat, Old_A_seurat_filtered, Old_A_seurat_norm, Old_B_data, Old_B_dbl, Old_B_seurat, Old_B_seurat_filtered, Old_B_seurat_norm, Old_C_data, Old_C_dbl, Old_C_seurat, Old_C_seurat_filtered, Old_C_seurat_norm,Old_D_data, Old_D_dbl, Old_D_seurat, Old_D_seurat_filtered, Old_D_seurat_norm)
```


## Merge
```{r, include=FALSE}
DefaultAssay(Young_A_singlets) <- "RNA"
DefaultAssay(Young_B_singlets) <- "RNA"
DefaultAssay(Young_C_singlets) <- "RNA"
DefaultAssay(Old_A_singlets) <- "RNA"
DefaultAssay(Old_B_singlets) <- "RNA"
DefaultAssay(Old_C_singlets) <- "RNA"
DefaultAssay(Old_D_singlets) <- "RNA"
DefaultAssay(Ger_A_singlets) <- "RNA"
DefaultAssay(Ger_B_singlets) <- "RNA"
DefaultAssay(Ger_C_singlets) <- "RNA"
DefaultAssay(Ger_D_singlets) <- "RNA"

Young_A_singlets$age <- "Young"
Young_B_singlets$age <- "Young"
Young_C_singlets$age <- "Young"
Old_A_singlets$age <- "Old"
Old_B_singlets$age <- "Old"
Old_C_singlets$age <- "Old"
Old_D_singlets$age <- "Old"
Ger_A_singlets$age <- "Geriatric"
Ger_B_singlets$age <- "Geriatric"
Ger_C_singlets$age <- "Geriatric"
Ger_D_singlets$age <- "Geriatric"

Young_A_singlets$orig.ident <- "Young_A"
Young_B_singlets$orig.ident <- "Young_B"

age_combined <- merge(Young_A_singlets, y = list(Young_B_singlets, Young_C_singlets, Old_A_singlets, Old_B_singlets, Old_C_singlets, Old_D_singlets, Ger_A_singlets, Ger_B_singlets, Ger_C_singlets, Ger_D_singlets), add.cell.ids = c("Young_A", "Young_B", "Young_C", "Old_A", "Old_B", "Old_C", "Old_D", "Ger_A", "Ger_B", "Ger_C", "Ger_D"))

age_combined$orig.ident <- factor(age_combined$orig.ident, levels = c("Young_A", "Young_B", "Young_C", "Old_A", "Old_B", "Old_C", "Old_D", "Ger_A", "Ger_B", "Ger_C", "Ger_D"))
```


## Pre-process

```{r, include=FALSE}
age_combined <- NormalizeData(age_combined) %>%
  FindVariableFeatures() %>%
  ScaleData() %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(age_combined, ndims = 50)
```


## Run harmony

```{r, include=FALSE}
age_combined <- RunHarmony(age_combined, group.by.vars = "age")

age_combined <- FindNeighbors(age_combined,reduction = "harmony",
                                dims = 1:40) %>% 
  FindClusters(resolution = 0.1)

age_combined <- RunUMAP(age_combined, reduction = "harmony", dims = 1:40)
```


## Order ages

```{r, include=FALSE}
age_combined$age <- factor(age_combined$age, levels = c("Young", "Old", "Geriatric"))
```


## Viualize data integration

```{r}
DimPlot(age_combined, reduction = "harmony", group.by = "age", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "harmony1", y = "harmony2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```



# 6. Annotate clusters

```{r}
DefaultAssay(age_combined) <- "RNA"
Idents(age_combined) <- "seurat_clusters"

cluster_ids <- levels(age_combined$seurat_clusters)

DotPlot(object = age_combined, features = c("Pax7", "Myf5", "Acta1", "Cdh5", "Myl9", "Pdgfrb", "Pdgfra", "Tnmd", "Mmrn1", "Cd68", "Csf1r", "S100a9", "Ms4a1", "Nkg7", "Il7r", "Ptn", "Mpz", "Gypa")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, face = "italic")) +
  scale_color_gradientn(colors = c("red", "white", "blue")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, face = "italic"))
```

```{r, include=FALSE}
cluster_labels <- c("Endothelial cells", 
                 "FAPs", 
                 "Myonuclei", 
                 "Neural/glial/Schwann cells",
                 "vSMCs/pericytes",
                 "Tenocytes",
                 "Monocytes/macrophages",
                 "FAPs",
                 "MuSCs",
                 "Erythrocytes",
                 "Endothelial cells",
                 "Neutrophils/T cells/NK cells",
                 "B cells")

cluster_to_cell_type <- setNames(cluster_labels, cluster_ids)

age_combined$seurat_clusters <- cluster_labels[age_combined$seurat_clusters]

age_combined$seurat_clusters <- factor(age_combined$seurat_clusters, levels = c("MuSCs", 
            "Myonuclei", 
            "Endothelial cells",
            "vSMCs/pericytes",
            "FAPs",
            "Tenocytes",
            "Monocytes/macrophages",
            "B cells",
            "Neutrophils/T cells/NK cells",
            "Neural/glial/Schwann cells",
            "Erythrocytes"))

Idents(age_combined) <- age_combined$seurat_clusters

age_combined <- JoinLayers(age_combined, what = "all")
```



# 8. Subset MuSCs

```{r, include=FALSE}
Idents(age_combined) <- "seurat_clusters"

age_MuSCs <- subset(age_combined, idents = "MuSCs")
```


## DEGs per condition

```{r, include=FALSE}
Idents(age_MuSCs) <- "age"

age_MuSCs_markers <- FindAllMarkers(object = age_MuSCs, group.by = "age", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```



# Cell scoring

## Create gene list
```{r, include=FALSE}
cachexia_gene_list <- cachexia_gene_list <- read.csv("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/MuSCs_combined/DEGs_MuSCs_combined_subclusters.csv")

cachexia_gene_list <- cachexia_gene_list[cachexia_gene_list$cluster == "Cachectic" & cachexia_gene_list$p_val_adj < 0.05, ]

cachexia_gene_list <- cachexia_gene_list$gene
```


## AddModuleScore by age

```{r, include=FALSE}
cachexia_score <- AddModuleScore_UCell(age_MuSCs, features = list(cachexia_gene_list), name = "cachexia_score")

means_cachexia_score <- cachexia_score@meta.data %>%
  group_by(age) %>%
  summarise(average_cachexia_score = mean(signature_1cachexia_score, na.rm = TRUE))
```

```{r}
print(means_cachexia_score)
```


```{r}
VlnPlot(cachexia_score, features = "signature_1cachexia_score", group.by = "age") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45)) +
  labs(x = NULL, y = "Cachexia Score", title = NULL) +
  geom_hline(yintercept = 0, linetype = "solid", color = "black")
```
