---
title: "LLC_timecourse_processing"
author: "Alex Brown"
date: "2025-06-18"
output: html_document
---



# Load packages

```{r, include=FALSE}
library(Seurat)
library(SeuratWrappers)
library(tidyverse)
library(Biobase)
library(BiocGenerics)
library(ggplot2)
library(cowplot)
library(patchwork)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(ggbeeswarm)
library(pheatmap)
library(viridis)
library(EnhancedVolcano)
library(scDblFinder)
```



# Read in the data

```{r, include=FALSE}
sham_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/data/sham_filtered_feature_bc_matrix")

two_weeks_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/data/2weeks_filtered_feature_bc_matrix")

two.five_weeks_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/data/2.5weeks_filtered_feature_bc_matrix")

three.five_weeks_data <- Read10X("C:/Users/nwb_l/Documents/r_projects/LLC_timecourse/data/3.5weeks_filtered_feature_bc_matrix")
```



# Create Seurat Object

```{r, include=FALSE}
sham_data_seurat <- CreateSeuratObject(counts = sham_data, min.cells = 3, project = "Sham")
two_weeks_data_seurat <- CreateSeuratObject(counts = two_weeks_data, min.cells = 3, project = "2_weeks")
two.five_weeks_data_seurat <- CreateSeuratObject(counts = two.five_weeks_data, min.cells = 3, project = "2.5_weeks")
three.five_weeks_data_seurat <- CreateSeuratObject(counts = three.five_weeks_data, min.cells = 3, project = "3.5_weeks")
```



# 1. QC

## % mitochondrial reads

```{r, include=FALSE}
sham_data_seurat[["percent.mito"]] <- PercentageFeatureSet(sham_data_seurat, pattern = "^mt-")
two_weeks_data_seurat[["percent.mito"]] <- PercentageFeatureSet(two_weeks_data_seurat, pattern = "^mt-")
two.five_weeks_data_seurat[["percent.mito"]] <- PercentageFeatureSet(two.five_weeks_data_seurat, pattern = "^mt-")
three.five_weeks_data_seurat[["percent.mito"]] <- PercentageFeatureSet(three.five_weeks_data_seurat, pattern = "^mt-")
```


## QC plots

### % mitochondrial reads

```{r, include=FALSE}
percent_mito_Sham <- VlnPlot(sham_data_seurat, features = "percent.mito") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5)) +
  labs(x = NULL, y = "% mitochondrial genes", title = NULL)

percent_mito_2_weeks <- VlnPlot(two_weeks_data_seurat, features = "percent.mito") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2 weeks")

percent_mito_2.5_weeks <- VlnPlot(two.five_weeks_data_seurat, features = "percent.mito") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2.5 weeks")

percent_mito_3.5_weeks <- VlnPlot(three.five_weeks_data_seurat, features = "percent.mito") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "3.5 weeks")
```

```{r}
percent_mito_Sham + percent_mito_2_weeks + percent_mito_2.5_weeks + percent_mito_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))
```

### nFeatures

```{r, include=FALSE}
nFeature_Sham <- VlnPlot(sham_data_seurat, features = "nFeature_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5)) +
  labs(x = NULL, y = "# features", title = NULL) +
  scale_y_continuous(limits = c(0, 10000))

nFeature_2_weeks <- VlnPlot(two_weeks_data_seurat, features = "nFeature_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2 weeks") +
  scale_y_continuous(limits = c(0, 10000))

nFeature_2.5_weeks <- VlnPlot(two.five_weeks_data_seurat, features = "nFeature_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2.5 weeks") +
  scale_y_continuous(limits = c(0, 10000))

nFeature_3.5_weeks <- VlnPlot(three.five_weeks_data_seurat, features = "nFeature_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "3.5 weeks") +
  scale_y_continuous(limits = c(0, 10000))
```

```{r}
nFeature_Sham + nFeature_2_weeks + nFeature_2.5_weeks + nFeature_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))
```

### nCounts
```{r, include=FALSE}
nCount_Sham <- VlnPlot(sham_data_seurat, features = "nCount_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5)) +
  labs(x = NULL, y = "# counts", title = NULL) +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nCount_2_weeks <- VlnPlot(two_weeks_data_seurat, features = "nCount_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2 weeks") +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nCount_2.5_weeks <- VlnPlot(two.five_weeks_data_seurat, features = "nCount_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "2.5 weeks") +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nCount_3.5_weeks <- VlnPlot(three.five_weeks_data_seurat, features = "nCount_RNA") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
  labs(x = NULL, y = NULL, title = NULL) +
  scale_x_discrete(labels = "3.5 weeks") +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))
```

```{r}
nCount_Sham + nCount_2_weeks + nCount_2.5_weeks + nCount_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))
```

### nFeatures vs. nCounts

```{r, include=FALSE}
nFeature_nCount_Sham <- FeatureScatter(sham_data_seurat, feature1 = "nFeature_RNA", feature2 = "nCount_RNA") +
  geom_smooth(method = 'lm') +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5, size = 5),
        axis.text.y = element_text(size = 5), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 15), plot.title = element_text(size = 10)) +
  labs(x = "Sham", y = "# counts") +
  scale_x_continuous(limits = c(0, 10000), breaks = seq(0, 10000, by = 2500)) +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nFeature_nCount_2_weeks <- FeatureScatter(two_weeks_data_seurat, feature1 = "nFeature_RNA", feature2 = "nCount_RNA") +
  geom_smooth(method = 'lm') +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5, size = 5), axis.text.y = element_blank(), axis.title.x = element_text(size = 10), plot.title = element_text(size = 10)) +
  labs(x = "2 weeks", y = NULL) +
  scale_x_continuous(limits = c(0, 10000), breaks = seq(0, 10000, by = 2500)) +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nFeature_nCount_2.5_weeks <- FeatureScatter(two.five_weeks_data_seurat, feature1 = "nFeature_RNA", feature2 = "nCount_RNA") +
  geom_smooth(method = 'lm') +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5, size = 5), axis.text.y = element_blank(), axis.title.x = element_text(size = 10), plot.title = element_text(size = 10)) +
  labs(x = "2.5 weeks", y = NULL) +
  scale_x_continuous(limits = c(0, 10000), breaks = seq(0, 10000, by = 2500)) +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))

nFeature_nCount_3.5_weeks <- FeatureScatter(three.five_weeks_data_seurat, feature1 = "nFeature_RNA", feature2 = "nCount_RNA") +
  geom_smooth(method = 'lm') +
  theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5, size = 5), axis.text.y = element_blank(), axis.title.x = element_text(size = 10), plot.title = element_text(size = 10)) +
  labs(x = "3.5 weeks", y = NULL) +
  scale_x_continuous(limits = c(0, 10000), breaks = seq(0, 10000, by = 2500)) +
  scale_y_continuous(limits = c(0, 100000), breaks = seq(0, 100000, by = 25000))
```

```{r}
nFeature_nCount_combined <- nFeature_nCount_Sham + nFeature_nCount_2_weeks + nFeature_nCount_2.5_weeks + nFeature_nCount_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))

nFeature_nCount_combined +
  plot_annotation(
    theme = theme(
      plot.caption = element_text(size = 15, hjust = 0.5),
      plot.margin = margin(t = 10, r = 10, b = 30, l = 10)), caption = "# features")
```



# 2. Filtering

```{r, include=FALSE}
sham_data_seurat_filtered <- subset(sham_data_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
two_weeks_data_seurat_filtered <- subset(two_weeks_data_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
two.five_weeks_data_seurat_filtered <- subset(two.five_weeks_data_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
three.five_weeks_data_seurat_filtered <- subset(three.five_weeks_data_seurat, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
```



# 3. Pre-process

```{r, include=FALSE}
all_genes_sham <- rownames(sham_data_seurat_filtered)
all_genes_two_weeks <- rownames(two_weeks_data_seurat_filtered)
all_genes_two.five_weeks <- rownames(two.five_weeks_data_seurat_filtered)
all_genes_three.five_weeks <- rownames(three.five_weeks_data_seurat_filtered)
```


## Sham

```{r, include=FALSE}
sham_data_seurat_norm <- sham_data_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_sham) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(sham_data_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
sham_data_seurat_norm <- sham_data_seurat_norm %>%
  FindNeighbors(dims = 1:38) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:38)
```


## 2 weeks

```{r, include=FALSE}
two_weeks_data_seurat_norm <- two_weeks_data_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_sham) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(two_weeks_data_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
two_weeks_data_seurat_norm <- two_weeks_data_seurat_norm %>%
  FindNeighbors(dims = 1:39) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:39)
```


## 2.5 weeks

```{r, include=FALSE}
two.five_weeks_data_seurat_norm <- two.five_weeks_data_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_sham) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(two.five_weeks_data_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
two.five_weeks_data_seurat_norm <- two.five_weeks_data_seurat_norm %>%
  FindNeighbors(dims = 1:35) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:35)
```


## 3.5 weeks

```{r, include=FALSE}
three.five_weeks_data_seurat_norm <- three.five_weeks_data_seurat_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_sham) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(three.five_weeks_data_seurat_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
three.five_weeks_data_seurat_norm <- three.five_weeks_data_seurat_norm %>%
  FindNeighbors(dims = 1:34) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:34)
```



# 4. scDblFinder

```{r, include=FALSE}
sham_dbl <- scDblFinder(GetAssayData(sham_data_seurat_norm, slot = "counts"), clusters = Idents(sham_data_seurat_norm))

sham_data_seurat_norm$scDblFinder.class <- sham_dbl$scDblFinder.class


two_weeks_dbl <- scDblFinder(GetAssayData(two_weeks_data_seurat_norm, slot = "counts"), clusters = Idents(two_weeks_data_seurat_norm))

two_weeks_data_seurat_norm$scDblFinder.class <- two_weeks_dbl$scDblFinder.class


two.five_weeks_dbl <- scDblFinder(GetAssayData(two.five_weeks_data_seurat_norm, slot = "counts"), clusters = Idents(two.five_weeks_data_seurat_norm))

two.five_weeks_data_seurat_norm$scDblFinder.class <- two.five_weeks_dbl$scDblFinder.class


three.five_weeks_dbl <- scDblFinder(GetAssayData(three.five_weeks_data_seurat_norm, slot = "counts"), clusters = Idents(three.five_weeks_data_seurat_norm))

three.five_weeks_data_seurat_norm$scDblFinder.class <- three.five_weeks_dbl$scDblFinder.class
```


## See numbers of singlets vs. doublets

```{r}
table(sham_data_seurat_norm$scDblFinder.class)
table(two_weeks_data_seurat_norm$scDblFinder.class)
table(two.five_weeks_data_seurat_norm$scDblFinder.class)
table(three.five_weeks_data_seurat_norm$scDblFinder.class)
```


## Visualize Doublets

```{r}
DimPlot(
  sham_data_seurat_norm,
  reduction = "umap",
  group.by = "scDblFinder.class",
  cells.highlight = list("doublet" = WhichCells(sham_data_seurat_norm, expression = scDblFinder.class == "doublet")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#E87D72"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  two_weeks_data_seurat_norm,
  reduction = "umap",
  group.by = "scDblFinder.class",
  cells.highlight = list("doublet" = WhichCells(two_weeks_data_seurat_norm, expression = scDblFinder.class == "doublet")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#87AD34"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  two.five_weeks_data_seurat_norm,
  reduction = "umap",
  group.by = "scDblFinder.class",
  cells.highlight = list("doublet" = WhichCells(two.five_weeks_data_seurat_norm, expression = scDblFinder.class == "doublet")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#56BCC2"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  three.five_weeks_data_seurat_norm,
  reduction = "umap",
  group.by = "scDblFinder.class",
  cells.highlight = list("doublet" = WhichCells(three.five_weeks_data_seurat_norm, expression = scDblFinder.class == "doublet")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#BC7FF8"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")
```

```{r, include=FALSE}
Idents(sham_data_seurat_norm) <- sham_data_seurat_norm@meta.data$scDblFinder.class

Idents(two_weeks_data_seurat_norm) <- two_weeks_data_seurat_norm@meta.data$scDblFinder.class

Idents(two.five_weeks_data_seurat_norm) <- two.five_weeks_data_seurat_norm@meta.data$scDblFinder.class

Idents(three.five_weeks_data_seurat_norm) <- three.five_weeks_data_seurat_norm@meta.data$scDblFinder.class
```

```{r, include=FALSE}
nFeature_doublets_Sham <- VlnPlot(sham_data_seurat_norm, features = "nFeature_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5)) +
    labs(x = "Sham", y = "# features", title = NULL) +
    scale_y_continuous(limits = c(0, 5000))

nFeature_doublets_2_weeks <- VlnPlot(two_weeks_data_seurat_norm, features = "nFeature_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "2 weeks", y = NULL, title = NULL) +    
    scale_y_continuous(limits = c(0, 5000))

nFeature_doublets_2.5_weeks <- VlnPlot(two.five_weeks_data_seurat_norm, features = "nFeature_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "2.5 weeks", y = NULL, title = NULL) +
    scale_y_continuous(limits = c(0, 5000))

nFeature_doublets_3.5_weeks <- VlnPlot(three.five_weeks_data_seurat_norm, features = "nFeature_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "3.5 weeks", y = NULL, title = NULL) +
    scale_y_continuous(limits = c(0, 5000))
```

```{r}
nFeature_doublets_Sham + nFeature_doublets_2_weeks + nFeature_doublets_2.5_weeks + nFeature_doublets_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))
```


```{r, include=FALSE}
nCount_doublets_Sham <- VlnPlot(sham_data_seurat_norm, features = "nCount_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5)) +
    labs(x = "Sham", y = "# Counts", title = NULL) +
    scale_y_continuous(limits = c(0, 15000))

nCount_doublets_2_weeks <- VlnPlot(two_weeks_data_seurat_norm, features = "nCount_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "2 weeks", y = NULL, title = NULL) +    
    scale_y_continuous(limits = c(0, 15000))

nCount_doublets_2.5_weeks <- VlnPlot(two.five_weeks_data_seurat_norm, features = "nCount_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "2.5 weeks", y = NULL, title = NULL) +
    scale_y_continuous(limits = c(0, 15000))

nCount_doublets_3.5_weeks <- VlnPlot(three.five_weeks_data_seurat_norm, features = "nCount_RNA", cols = c("singlet" = "#4575B4", "doublet" = "#D73027")) + theme(legend.position = "none", axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5), axis.text.y = element_blank()) +
    labs(x = "3.5 weeks", y = NULL, title = NULL) +
    scale_y_continuous(limits = c(0, 15000))
```

```{r}
nCount_doublets_Sham + nCount_doublets_2_weeks + nCount_doublets_2.5_weeks + nCount_doublets_3.5_weeks +
  plot_layout(ncol = 4, widths = c(1, 1, 1, 1))
```


## Remove Doublets

```{r, include=FALSE}
sham_singlets <- subset(sham_data_seurat_norm, subset = scDblFinder.class == "singlet")

two_weeks_singlets <- subset(two_weeks_data_seurat_norm, subset = scDblFinder.class == "singlet")

two.five_weeks_singlets <- subset(two.five_weeks_data_seurat_norm, subset = scDblFinder.class == "singlet")

three.five_weeks_singlets <- subset(three.five_weeks_data_seurat_norm, subset = scDblFinder.class == "singlet")
```



# 5. Combine conditions

## Merge data

```{r, include=FALSE}
ordered_datasets <- c(sham_singlets, two_weeks_singlets, two.five_weeks_singlets, three.five_weeks_singlets)

features_merge <- SelectIntegrationFeatures(object.list = ordered_datasets)
```


## Find anchors

```{r, include=FALSE}
anchors_merge <- FindIntegrationAnchors(object.list = list(sham_singlets, two_weeks_singlets, two.five_weeks_singlets, three.five_weeks_singlets), anchor.features = features_merge)
```


## Combine data

```{r, include=FALSE}
LLC_combined <- IntegrateData(anchorset = anchors_merge)

DefaultAssay(LLC_combined) <- "integrated"
```



# 6. Reprocess

## Scale data and run PCA
```{r, include=FALSE}
all_genes <- rownames(LLC_combined)
LLC_combined <- ScaleData(LLC_combined, features = all_genes)
LLC_combined <- RunPCA(LLC_combined, npcs = 50)
```

```{r}
ElbowPlot(LLC_combined, ndims = 50) ### to check dims
```


## Cluster

```{r, include=FALSE}
LLC_combined <- LLC_combined %>%
  FindNeighbors(dims = 1:42) %>%
  FindClusters(resolution = 0.1)

Idents(LLC_combined) <- "seurat_clusters"

umap.method = 'umap-learn'
metric = 'correlation'

LLC_combined <- RunUMAP(LLC_combined, reduction = "pca", dims = 1:42)
```



# 7. Annotate cells

```{r, include=FALSE}
cluster_ids <- levels(LLC_combined$seurat_clusters)
```

```{r, include=FALSE}
cluster_labels <- c("Endothelial cells", 
                 "Myonuclei", 
                 "FAPs", 
                 "MuSCs",
                 "Monocytes/macrophages",
                 "Neutrophils",
                 "Erythrocytes",
                 "Endothelial cells",
                 "Pericytes",
                 "vSMCs",
                 "B cells",
                 "Schwann cells",
                 "Erythrocytes",
                 "Endothelial cells",
                 "Neural/glial cells",
                 "T cells/NK cells",
                 "Platelets",
                 "vSMCs",
                 "FAPs",
                 "Epithelial cells")
```

```{r, include=FALSE}
cluster_to_cell_type <- setNames(cluster_labels, cluster_ids)

LLC_combined$seurat_clusters <- cluster_labels[LLC_combined$seurat_clusters]
```


## Order conditions

```{r, include=FALSE}
LLC_combined$orig.ident <- factor(LLC_combined$orig.ident, levels = c("Sham", "2_weeks", "2.5_weeks", "3.5_weeks"))
```


## Order clusters

```{r, include=FALSE}
LLC_combined$seurat_clusters <- factor(LLC_combined$seurat_clusters, levels = c("MuSCs", 
            "Myonuclei", 
            "Endothelial cells",
            "vSMCs",
            "Pericytes",
            "FAPs",
            "Platelets",
            "Monocytes/macrophages",
            "Neutrophils",
            "B cells",
            "T cells/NK cells",
            "Neural/glial cells",
            "Schwann cells",
            "Erythrocytes",
            "Epithelial cells"))
```

### DotPlot to show differentially expressed genes between clusters

```{r}
DefaultAssay(LLC_combined) <- "RNA"
Idents(LLC_combined) <- "seurat_clusters"

DotPlot(object = LLC_combined, features = c("Pax7", "Myf5", "Acta1", "Tnnc2", "Ptprb", "Cdh5", "Myl9", "Myh11", "Abcc9", "Pdgfrb", "Pdgfra", "Igfbp6", "Mmrn1", "Ccl21a", "Cd68", "Csf1r", "S100a8", "S100a9", "Cd79a", "Ms4a1", "Nkg7", "Il7r", "Ptn", "Cdh19", "Mpz", "S100b")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, face = "italic")) +
  scale_color_gradientn(colors = c("red", "white", "blue")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, face = "italic"))
```

### Cell numbers per cluster for each time point

```{r}
cell_counts_cluster <- table(LLC_combined@meta.data$seurat_clusters, LLC_combined@meta.data$orig.ident)
```

```{r, include=FALSE}
write.csv(cell_counts_cluster, file = "cell_counts_cluster.csv", row.names = TRUE)
```


## Remove Erythrocytes and Epithelial cells

```{r, include=FALSE}
erythrocytes_removed <- "Erythrocytes"

LLC_combined <- subset(LLC_combined, seurat_clusters != erythrocytes_removed)

epithelial_cells_removed <- "Epithelial cells"

LLC_combined <- subset(LLC_combined, seurat_clusters != epithelial_cells_removed)
```


## Visualize

### UMAP clusters

```{r}
DimPlot(LLC_combined, reduction = "umap", group.by = "seurat_clusters", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```

### UMAP conditions

```{r}
DimPlot(LLC_combined, reduction = "umap", group.by = "orig.ident", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```

### Highlighted by condition

```{r}
DimPlot(
  LLC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("Sham" = WhichCells(LLC_combined, expression = orig.ident == "Sham")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#E87D72"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  LLC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("2_weeks" = WhichCells(LLC_combined, expression = orig.ident == "2_weeks")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#87AD34"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  LLC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("2.5_weeks" = WhichCells(LLC_combined, expression = orig.ident == "2.5_weeks")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#56BCC2"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  LLC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("3.5_weeks" = WhichCells(LLC_combined, expression = orig.ident == "3.5_weeks")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#BC7FF8"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")
```

### Stacked barplot for cell counts per time point
#### Extract cluster proportions
```{r, include=FALSE}
cluster_proportions <- LLC_combined@meta.data %>%
  dplyr::select(orig.ident, seurat_clusters) %>%
  group_by(orig.ident, seurat_clusters) %>%
  summarize(count = n()) %>%
  group_by(orig.ident) %>%
  mutate(percentage = count / sum(count) * 100)
```

```{r}
write.csv(cluster_proportions, file = "cluster_proportions.csv", row.names = TRUE)
```

#### Create stacked barplot
```{r}
cluster_proportions$orig.ident <- factor(
  cluster_proportions$orig.ident,
  levels = c("Sham", "2_weeks", "2.5_weeks", "3.5_weeks"),
  labels = c("Sham", "2 weeks", "2.5 weeks", "3.5 weeks"))

ggplot(cluster_proportions, aes(x = orig.ident, y = percentage, fill = seurat_clusters)) +
  geom_bar(stat = "identity", width = 0.75, color = "black") +
  scale_y_continuous(labels = scales::percent_format(scale = 1), expand = c(0, 0)) +
  labs(title = "", x = "", y = "Proportion") +
  theme(legend.position = "right",  panel.background = element_blank(), axis.line = element_line(color = "black"), axis.text.x = element_text(angle = 45, hjust = 1)) +
  guides(fill = guide_legend(title = NULL))
```


## Clean environment (remove things that are no longer needed)

```{r, include=FALSE}
rm(anchors_merge, sham_data, sham_data_seurat, sham_data_seurat_filtered, sham_data_seurat_norm, sham_dbl, sham_singlets, two_weeks_data, two_weeks_data_seurat, two_weeks_data_seurat_filtered, two_weeks_data_seurat_norm, two_weeks_dbl, two_weeks_singlets, two.five_weeks_data, two.five_weeks_data_seurat, two.five_weeks_data_seurat_filtered, two.five_weeks_data_seurat_norm, two.five_weeks_dbl, two.five_weeks_singlets, three.five_weeks_data, three.five_weeks_data_seurat, three.five_weeks_data_seurat_filtered, three.five_weeks_data_seurat_norm, three.five_weeks_dbl, three.five_weeks_singlets, nCount_Sham, nCount_2_weeks, nCount_2.5_weeks, nCount_3.5_weeks, nCount_combined, nFeature_Sham, nFeature_2_weeks, nFeature_2.5_weeks, nFeature_3.5_weeks, nFeature_combined, nFeature_nCount_Sham, nFeature_nCount_2_weeks, nFeature_nCount_2.5_weeks, nFeature_nCount_3.5_weeks, nFeature_nCount_combined, nFeature_nCount_combined_labeled, percent_mito_Sham, percent_mito_2_weeks, percent_mito_2.5_weeks, percent_mito_3.5_weeks, percent_mito_combined)
```


## Find markers between clusters

```{r, include=FALSE}
DefaultAssay(LLC_combined) <- "RNA"

Idents(LLC_combined) <- LLC_combined$seurat_clusters

LLC_combined <- JoinLayers(LLC_combined, what = "all")

DEGs_LLC_clusters <- FindAllMarkers(object = LLC_combined, group.by = "seurat_clusters", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r, include=FALSE}
write.csv(DEGs_LLC_clusters, file = "DEGs_LLC_clusters.csv", row.names = FALSE)
```



# 8. Subset

## Conditions

```{r, include=FALSE}
Idents(LLC_combined) <- "orig.ident"

Sham <- subset(LLC_combined, idents = "Sham")
two_weeks <- subset(LLC_combined, idents = "2_weeks")
two.five_weeks <- subset(LLC_combined, idents = "2.5_weeks")
three.five_weeks <- subset(LLC_combined, idents = "3.5_weeks")
```


## Clusters

```{r, include=FALSE}
Idents(LLC_combined) <- "seurat_clusters"

LLC_MuSCs <- subset(LLC_combined, idents = "MuSCs")
LLC_Myonuclei <- subset(LLC_combined, idents = "Myonuclei")
LLC_Endothelial_cells <- subset(LLC_combined, idents = "Endothelial cells")
LLC_vSMCs <- subset(LLC_combined, idents = "vSMCs")
LLC_Pericytes <- subset(LLC_combined, idents = "Pericytes")
LLC_FAPs <- subset(LLC_combined, idents = "FAPs")
LLC_Platelets <- subset(LLC_combined, idents = "Platelets")
LLC_Monocytes_macrophages <- subset(LLC_combined, idents = "Monocytes/macrophages")
LLC_Neutrophils <- subset(LLC_combined, idents = "Neutrophils")
LLC_B_cells <- subset(LLC_combined, idents = "B cells")
LLC_T_NK_cells <- subset(LLC_combined, idents = "T cells/NK cells")
LLC_Neural_glial_cells <- subset(LLC_combined, idents = "Neural/glial cells")
LLC_Schwann_cells <- subset(LLC_combined, idents = "Schwann cells")
```


## DEGs per condition for each cluster

```{r, include=FALSE}
Idents(LLC_MuSCs) <- "orig.ident"
Idents(LLC_Myonuclei) <- "orig.ident"
Idents(LLC_Endothelial_cells) <- "orig.ident"
Idents(LLC_vSMCs) <- "orig.ident"
Idents(LLC_Pericytes) <- "orig.ident"
Idents(LLC_FAPs) <- "orig.ident"
Idents(LLC_Platelets) <- "orig.ident"
Idents(LLC_Monocytes_macrophages) <- "orig.ident"
Idents(LLC_Neutrophils) <- "orig.ident"
Idents(LLC_B_cells) <- "orig.ident"
Idents(LLC_T_NK_cells) <- "orig.ident"
Idents(LLC_Neural_glial_cells) <- "orig.ident"
Idents(LLC_Schwann_cells) <- "orig.ident"

DEGs_LLC_MuSCs <- FindAllMarkers(object = LLC_MuSCs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Myonuclei <- FindAllMarkers(object = LLC_Myonuclei, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Endothelial_cells <- FindAllMarkers(object = LLC_Endothelial_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_vSMCs <- FindAllMarkers(object = LLC_vSMCs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Pericytes <- FindAllMarkers(object = LLC_Pericytes, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_FAPs <- FindAllMarkers(object = LLC_FAPs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Platelets <- FindAllMarkers(object = LLC_Platelets, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Monocytes_macrophages <- FindAllMarkers(object = LLC_Monocytes_macrophages, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Neutrophils <- FindAllMarkers(object = LLC_Neutrophils, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_B_cells <- FindAllMarkers(object = LLC_B_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_T_NK_cells <- FindAllMarkers(object = LLC_T_NK_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Neural_glial_cells <- FindAllMarkers(object = LLC_Neural_glial_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_LLC_Schwann_cells <- FindAllMarkers(object = LLC_Schwann_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r, include=FALSE}
write.csv(DEGs_LLC_MuSCs, file = "DEGs_LLC_MuSCs.csv", row.names = FALSE)

write.csv(DEGs_LLC_Myonuclei, file = "DEGs_LLC_Myonuclei.csv", row.names = FALSE)

write.csv(DEGs_LLC_Endothelial_cells, file = "DEGs_LLC_Endothelial_cells.csv", row.names = FALSE)

write.csv(DEGs_LLC_vSMCs, file = "DEGs_LLC_vSMCs.csv", row.names = FALSE)

write.csv(DEGs_LLC_Pericytes, file = "DEGs_LLC_Pericytes.csv", row.names = FALSE)

write.csv(DEGs_LLC_FAPs, file = "DEGs_LLC_FAPs.csv", row.names = FALSE)

write.csv(DEGs_LLC_Platelets, file = "DEGs_LLC_Platelets.csv", row.names = FALSE)

write.csv(DEGs_LLC_Monocytes_macrophages, file = "DEGs_LLC_Monocytes_macrophages.csv", row.names = FALSE)

write.csv(DEGs_LLC_Neutrophils, file = "DEGs_LLC_Neutrophils.csv", row.names = FALSE)

write.csv(DEGs_LLC_B_cells, file = "DEGs_LLC_B_cells.csv", row.names = FALSE)

write.csv(DEGs_LLC_T_NK_cells, file = "DEGs_LLC_T__NK_cells.csv", row.names = FALSE)

write.csv(DEGs_LLC_Neural_glial_cells, file = "DEGs_LLC_Neural_glial_cells.csv", row.names = FALSE)

write.csv(DEGs_LLC_Schwann_cells, file = "DEGs_LLC_Schwann_cells.csv", row.names = FALSE)
```


## Remove unused objects

```{r, include=FALSE}
rm(cluster_proportions, ordered_datasets, all_genes, all_genes_sham, all_genes_two_weeks, all_genes_two.five_weeks, all_genes_three.five_weeks, cell_counts_cluster, cluster_ids, cluster_labels, cluster_to_cell_type, epithelial_cells_removed, erythrocytes_removed, features_merge, metric, umap.method)
```



# Heatmaps for top DEGs per condition for each cluster

```{r, include=FALSE}
DefaultAssay(LLC_MuSCs) <- "RNA"
DefaultAssay(LLC_Myonuclei) <- "RNA"
DefaultAssay(LLC_Endothelial_cells) <- "RNA"
DefaultAssay(LLC_vSMCs) <- "RNA"
DefaultAssay(LLC_Pericytes) <- "RNA"
DefaultAssay(LLC_FAPs) <- "RNA"
DefaultAssay(LLC_Platelets) <- "RNA"
DefaultAssay(LLC_Monocytes_macrophages) <- "RNA"
DefaultAssay(LLC_Neutrophils) <- "RNA"
DefaultAssay(LLC_B_cells) <- "RNA"
DefaultAssay(LLC_T_NK_cells) <- "RNA"
DefaultAssay(LLC_Neural_glial_cells) <- "RNA"
DefaultAssay(LLC_Schwann_cells) <- "RNA"
```


## Scale data

```{r, include=FALSE}
LLC_MuSCs <- ScaleData(LLC_MuSCs, features = rownames(LLC_MuSCs))
LLC_Myonuclei <- ScaleData(LLC_Myonuclei, features = rownames(LLC_Myonuclei))
LLC_Endothelial_cells <- ScaleData(LLC_Endothelial_cells, features = rownames(LLC_Endothelial_cells))
LLC_vSMCs <- ScaleData(LLC_vSMCs, features = rownames(LLC_vSMCs))
LLC_Pericytes <- ScaleData(LLC_Pericytes, features = rownames(LLC_Pericytes))
LLC_FAPs <- ScaleData(LLC_FAPs, features = rownames(LLC_FAPs))
LLC_Platelets <- ScaleData(LLC_Platelets, features = rownames(LLC_Platelets))
LLC_Monocytes_macrophages <- ScaleData(LLC_Monocytes_macrophages, features = rownames(LLC_Monocytes_macrophages))
LLC_Neutrophils <- ScaleData(LLC_Neutrophils, features = rownames(LLC_Neutrophils))
LLC_B_cells <- ScaleData(LLC_B_cells, features = rownames(LLC_B_cells))
LLC_T_NK_cells <- ScaleData(LLC_T_NK_cells, features = rownames(LLC_T_NK_cells))
LLC_Neural_glial_cells <- ScaleData(LLC_Neural_glial_cells, features = rownames(LLC_Neural_glial_cells))
LLC_Schwann_cells <- ScaleData(LLC_Schwann_cells, features = rownames(LLC_Schwann_cells))
```


## Top 5 DE genes by condition
### NOTE - "cluster" is actually "orig.ident" because that is what I specified with FindAllMarkers

```{r, include=FALSE}
top5_LLC_MuSCs <- DEGs_LLC_MuSCs %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Myonuclei <- DEGs_LLC_Myonuclei %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Endothelial_cells <- DEGs_LLC_Endothelial_cells %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_vSMCs <- DEGs_LLC_vSMCs %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Pericytes <- DEGs_LLC_Pericytes %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_FAPs <- DEGs_LLC_FAPs %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Platelets <- DEGs_LLC_Platelets %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Monocytes_macrophages <- DEGs_LLC_Monocytes_macrophages %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Neutrophils <- DEGs_LLC_Neutrophils %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_B_cells <- DEGs_LLC_B_cells %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_T_NK_cells <- DEGs_LLC_T_NK_cells %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Neural_glial_cells <- DEGs_LLC_Neural_glial_cells %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
top5_LLC_Schwann_cells <- DEGs_LLC_Schwann_cells %>%
  group_by(cluster) %>% 
  top_n(n = 5, wt = avg_log2FC)
```


## Run Heatmap

```{r}
DoHeatmap(LLC_MuSCs, features = top5_LLC_MuSCs$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) + 
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Myonuclei, features = top5_LLC_Myonuclei$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Endothelial_cells, features = top5_LLC_Endothelial_cells$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_vSMCs, features = top5_LLC_vSMCs$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Pericytes, features = top5_LLC_Pericytes$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_FAPs, features = top5_LLC_FAPs$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Platelets, features = top5_LLC_Platelets$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Monocytes_macrophages, features = top5_LLC_Monocytes_macrophages$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmapLLC_(Neutrophils, features = top5_LLC_Neutrophils$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_B_cells, features = top5_LLC_B_cells$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_T_NK_cells, features = top5_LLC_T_NK_cells$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Neural_glial_cells, features = top5_LLC_Neural_glial_cells$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))

DoHeatmap(LLC_Schwann_cells, features = top5_LLC_Schwann_cells$gene, draw.lines = TRUE, angle = 90, size = 3, raster = FALSE, label = FALSE) +
  scale_fill_gradientn(colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"), limits = c(-2.5, 2.5)) +
  theme(legend.position = "right", legend.key.size = unit(0.5, "lines"))
```



# Heatmap

## By cluster

```{r, include=FALSE}
DefaultAssay(LLC_combined) <- "RNA"
Idents(LLC_combined) <- "seurat_clusters"

top5_LLC_combined_cluster <- DEGs_LLC_clusters %>%
  group_by(cluster) %>%
  slice_max(order_by = avg_log2FC, n = 5)

top_genes <- unique(top5_LLC_combined_cluster$gene)

LLC_combined <- ScaleData(LLC_combined, features = top_genes)

LLC_combined_colors <- c(
  "MuSCs" = "#EA7F74",
  "Myonuclei" = "#D68E32",
  "Endothelial cells" = "#B79C32",
  "vSMCs" = "#91A934",
  "Pericytes" = "#57B133",
  "FAPs" = "#55BA76",
  "Platelets" = "#56BDAA",
  "Monocytes/macrophages" = "#55B7D4",
  "Neutrophils" = "#4CA9F5",
  "B cells" = "#8B92F8",
  "T cells/NK cells" = "#C67AF4",
  "Neural/glial cells" = "#E46BD6",
  "Schwann cells" = "#EB6EA8"
)
```

```{r}
DoHeatmap(
  LLC_combined,
  features = top5_LLC_combined_cluster$gene,
  draw.lines = TRUE,
  angle = 90,
  size = 3,
  label = FALSE,
  group.colors = LLC_combined_colors
) +
  scale_fill_gradientn(
    colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"),
    limits = c(-2.5, 2.5)
  ) +
  theme(
    legend.position = "right",
    legend.key.size = unit(0.5, "lines")
  )
```


## By condition

```{r, include=FALSE}
DefaultAssay(LLC_combined) <- "RNA"
Idents(LLC_combined) <- "orig.ident"

DEGs_LLC_condition <- FindAllMarkers(object = LLC_combined, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)

top5_LLC_condition <- DEGs_LLC_condition %>%
  group_by(cluster) %>%
  slice_max(order_by = avg_log2FC, n = 5)

top_genes_condition <- unique(top5_LLC_condition$gene)

LLC_combined <- ScaleData(LLC_combined, features = top_genes_condition)

LLC_condition_colors <- c(
  "Sham" = "#E87D72",
  "2_weeks" = "#87AD34",
  "2.5_weeks" = "#56BCC2",
  "3.5_weeks" = "#BC7FF8")
```

```{r}
DoHeatmap(
  LLC_combined,
  features = top5_LLC_condition$gene,
  draw.lines = TRUE,
  angle = 90,
  size = 3,
  raster = FALSE,
  label = FALSE,
  group.colors = LLC_condition_colors
) +
  scale_fill_gradientn(
    colors = c("#D73027", "#FEE090", "#FFFFFF", "#91BFDB", "#4575B4"),
    limits = c(-2.5, 2.5)
  ) +
  theme(
    legend.position = "right",
    legend.key.size = unit(0.5, "lines")
  )
```
