---
title: "Pryce_PDAC_processing"
author: "Alex Brown"
date: "2025-07-25"
output: html_document
---



# Load packages

```{r, include=FALSE}
library(Seurat)
library(SeuratWrappers)
library(hdf5r)
library(harmony)
library(tidyverse)
library(Biobase)
library(BiocGenerics)
library(ggplot2)
library(cowplot)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(ggbeeswarm)
library(pheatmap)
library(viridis)
library(EnhancedVolcano)
library(scDblFinder)
library(harmony)
```



# Read in data

```{r, include=FALSE}
getwd()
```

## CD45- control

```{r, include=FALSE}
PDAC_CD45neg_CTL <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920256_S72_raw_feature_bc_matrix.h5")
```


## CD45+ control

```{r, include=FALSE}
PDAC_CD45pos_CTL <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920257_S73_raw_feature_bc_matrix.h5")
```


## CD45- weight-stable PDAC

```{r, include=FALSE}
PDAC_CD45neg_WS <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920254_S70_raw_feature_bc_matrix.h5")
```


## CD45+ weight-stable PDAC

```{r, include=FALSE}
PDAC_CD45pos_WS <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920255_S71_raw_feature_bc_matrix.h5")
```


## CD45- cachectic PDAC

```{r, include=FALSE}
PDAC_CD45neg_cac <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920258_S74_raw_feature_bc_matrix.h5")
```


## CD45+ cachectic PDAC

```{r, include=FALSE}
PDAC_CD45pos_cac <- Read10X_h5("C:/Users/nwb_l/Documents/r_projects/Pryce_2025_scRNAseq/data/GSM7920259_S75_raw_feature_bc_matrix.h5")
```



# Create Seurat Object

```{r, include=FALSE}
PDAC_CD45neg_CTL_so <- CreateSeuratObject(counts = PDAC_CD45neg_CTL, min.cells = 3, project = "PDAC_CD45neg_CTL")

PDAC_CD45pos_CTL_so <- CreateSeuratObject(counts = PDAC_CD45pos_CTL, min.cells = 3, project = "PDAC_CD45pos_CTL")

PDAC_CD45neg_WS_so <- CreateSeuratObject(counts = PDAC_CD45neg_WS, min.cells = 3, project = "PDAC_CD45neg_WS")

PDAC_CD45pos_WS_so <- CreateSeuratObject(counts = PDAC_CD45pos_WS, min.cells = 3, project = "PDAC_CD45pos_WS")

PDAC_CD45neg_cac_so <- CreateSeuratObject(counts = PDAC_CD45neg_cac, min.cells = 3, project = "PDAC_CD45neg_cac")

PDAC_CD45pos_cac_so <- CreateSeuratObject(counts = PDAC_CD45pos_cac, min.cells = 3, project = "PDAC_CD45pos_cac")
```



# 1. QC

## % mitochondrial reads

```{r, include=FALSE}
PDAC_CD45neg_CTL_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45neg_CTL_so, pattern = "^mt-")

PDAC_CD45pos_CTL_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45pos_CTL_so, pattern = "^mt-")

PDAC_CD45neg_WS_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45neg_WS_so, pattern = "^mt-")

PDAC_CD45pos_WS_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45pos_WS_so, pattern = "^mt-")

PDAC_CD45neg_cac_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45neg_cac_so, pattern = "^mt-")

PDAC_CD45pos_cac_so[["percent.mito"]] <- PercentageFeatureSet(PDAC_CD45pos_cac_so, pattern = "^mt-")
```



# 2. Filtering

```{r, include=FALSE}
PDAC_CD45neg_CTL_filtered <- subset(PDAC_CD45neg_CTL_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)

PDAC_CD45pos_CTL_filtered <- subset(PDAC_CD45pos_CTL_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)

PDAC_CD45neg_WS_filtered <- subset(PDAC_CD45neg_WS_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)

PDAC_CD45pos_WS_filtered <- subset(PDAC_CD45pos_WS_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)

PDAC_CD45neg_cac_filtered <- subset(PDAC_CD45neg_cac_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)

PDAC_CD45pos_cac_filtered <- subset(PDAC_CD45pos_cac_so, subset = nFeature_RNA > 200 & nFeature_RNA < 60000 & nCount_RNA > 500 & nCount_RNA < 15000 & percent.mito < 10)
```



# 3. Pre-process

```{r, include=FALSE}
all_genes_PDAC_CD45neg_CTL <- rownames(PDAC_CD45neg_CTL_filtered)
all_genes_PDAC_CD45pos_CTL <- rownames(PDAC_CD45pos_CTL_filtered)
all_genes_PDAC_CD45neg_WS <- rownames(PDAC_CD45neg_WS_filtered)
all_genes_PDAC_CD45pos_WS <- rownames(PDAC_CD45pos_WS_filtered)
all_genes_PDAC_CD45neg_cac <- rownames(PDAC_CD45neg_cac_filtered)
all_genes_PDAC_CD45pos_cac <- rownames(PDAC_CD45pos_cac_filtered)
```


## PDAC_CD45neg_CTL

```{r, include=FALSE}
PDAC_CD45neg_CTL_norm <- PDAC_CD45neg_CTL_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45neg_CTL) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45neg_CTL_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45neg_CTL_norm <- PDAC_CD45neg_CTL_norm %>%
  FindNeighbors(dims = 1:38) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:38)
```


## PDAC_CD45pos_CTL

```{r, include=FALSE}
PDAC_CD45pos_CTL_norm <- PDAC_CD45pos_CTL_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45pos_CTL) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45pos_CTL_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45pos_CTL_norm <- PDAC_CD45pos_CTL_norm %>%
  FindNeighbors(dims = 1:39) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:39)
```


## PDAC_CD45neg_WS

```{r, include=FALSE}
PDAC_CD45neg_WS_norm <- PDAC_CD45neg_WS_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45neg_WS) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45neg_WS_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45neg_WS_norm <- PDAC_CD45neg_WS_norm %>%
  FindNeighbors(dims = 1:41) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:41)
```


## PDAC_CD45pos_WS

```{r, include=FALSE}
PDAC_CD45pos_WS_norm <- PDAC_CD45pos_WS_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45pos_WS) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45pos_WS_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45pos_WS_norm <- PDAC_CD45pos_WS_norm %>%
  FindNeighbors(dims = 1:43) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:43)
```


## PDAC_CD45neg_cac

```{r, include=FALSE}
PDAC_CD45neg_cac_norm <- PDAC_CD45neg_cac_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45neg_cac) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45neg_cac_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45neg_cac_norm <- PDAC_CD45neg_cac_norm %>%
  FindNeighbors(dims = 1:35) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:35)
```


## PDAC_CD45pos_cac

```{r, include=FALSE}
PDAC_CD45pos_cac_norm <- PDAC_CD45pos_cac_filtered %>%
  NormalizeData(normalization.method = "LogNormalize", scale.factor = 10000) %>%
  FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>%
  ScaleData(features = all_genes_PDAC_CD45pos_cac) %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_CD45pos_cac_norm, ndims = 50) ### to check dims
```

```{r, include=FALSE}
PDAC_CD45pos_cac_norm <- PDAC_CD45pos_cac_norm %>%
  FindNeighbors(dims = 1:39) %>%
  FindClusters(resolution = 0.1) %>%
  RunUMAP(dims = 1:39)
```



# 4. scDblFinder

```{r, include=FALSE}
## CTL

PDAC_CD45neg_CTL_dbl <- scDblFinder(GetAssayData(PDAC_CD45neg_CTL_norm, slot = "counts"), clusters = Idents(PDAC_CD45neg_CTL_norm))

PDAC_CD45neg_CTL_norm$scDblFinder.class <- PDAC_CD45neg_CTL_dbl$scDblFinder.class


PDAC_CD45pos_CTL_dbl <- scDblFinder(GetAssayData(PDAC_CD45pos_CTL_norm, slot = "counts"), clusters = Idents(PDAC_CD45pos_CTL_norm))

PDAC_CD45pos_CTL_norm$scDblFinder.class <- PDAC_CD45pos_CTL_dbl$scDblFinder.class


## WS

PDAC_CD45neg_WS_dbl <- scDblFinder(GetAssayData(PDAC_CD45neg_WS_norm, slot = "counts"), clusters = Idents(PDAC_CD45neg_WS_norm))

PDAC_CD45neg_WS_norm$scDblFinder.class <- PDAC_CD45neg_WS_dbl$scDblFinder.class


PDAC_CD45pos_WS_dbl <- scDblFinder(GetAssayData(PDAC_CD45pos_WS_norm, slot = "counts"), clusters = Idents(PDAC_CD45pos_WS_norm))

PDAC_CD45pos_WS_norm$scDblFinder.class <- PDAC_CD45pos_WS_dbl$scDblFinder.class


## cac

PDAC_CD45neg_cac_dbl <- scDblFinder(GetAssayData(PDAC_CD45neg_cac_norm, slot = "counts"), clusters = Idents(PDAC_CD45neg_cac_norm))

PDAC_CD45neg_cac_norm$scDblFinder.class <- PDAC_CD45neg_cac_dbl$scDblFinder.class


PDAC_CD45pos_cac_dbl <- scDblFinder(GetAssayData(PDAC_CD45pos_cac_norm, slot = "counts"), clusters = Idents(PDAC_CD45pos_cac_norm))

PDAC_CD45pos_cac_norm$scDblFinder.class <- PDAC_CD45pos_cac_dbl$scDblFinder.class
```


## See number of singlets vs. doublets

```{r}
table(PDAC_CD45neg_CTL_norm$scDblFinder.class)
table(PDAC_CD45pos_CTL_norm$scDblFinder.class)
table(PDAC_CD45neg_WS_norm$scDblFinder.class)
table(PDAC_CD45pos_WS_norm$scDblFinder.class)
table(PDAC_CD45neg_cac_norm$scDblFinder.class)
table(PDAC_CD45pos_cac_norm$scDblFinder.class)
```


## Remove doublets

```{r, include=FALSE}
PDAC_CD45neg_CTL_singlets <- subset(PDAC_CD45neg_CTL_norm, subset = scDblFinder.class == "singlet")

PDAC_CD45pos_CTL_singlets <- subset(PDAC_CD45pos_CTL_norm, subset = scDblFinder.class == "singlet")

PDAC_CD45neg_WS_singlets <- subset(PDAC_CD45neg_WS_norm, subset = scDblFinder.class == "singlet")

PDAC_CD45pos_WS_singlets <- subset(PDAC_CD45pos_WS_norm, subset = scDblFinder.class == "singlet")

PDAC_CD45neg_cac_singlets <- subset(PDAC_CD45neg_cac_norm, subset = scDblFinder.class == "singlet")

PDAC_CD45pos_cac_singlets <- subset(PDAC_CD45pos_cac_norm, subset = scDblFinder.class == "singlet")
```



# 5. Combine conditions

## Remove unused objects from environment to allow processing

```{r, include=FALSE}
rm(PDAC_CD45neg_CTL, PDAC_CD45neg_CTL_dbl, PDAC_CD45neg_CTL_filtered, PDAC_CD45neg_CTL_norm, PDAC_CD45neg_CTL_so, PDAC_CD45pos_CTL, PDAC_CD45pos_CTL_dbl, PDAC_CD45pos_CTL_filtered, PDAC_CD45pos_CTL_norm, PDAC_CD45pos_CTL_so, PDAC_CD45neg_WS, PDAC_CD45neg_WS_dbl, PDAC_CD45neg_WS_filtered, PDAC_CD45neg_WS_norm, PDAC_CD45neg_WS_so, PDAC_CD45pos_WS, PDAC_CD45pos_WS_dbl, PDAC_CD45pos_WS_filtered, PDAC_CD45pos_WS_norm, PDAC_CD45pos_WS_so, PDAC_CD45neg_cac, PDAC_CD45neg_cac_dbl, PDAC_CD45neg_cac_filtered, PDAC_CD45neg_cac_norm, PDAC_CD45neg_cac_so, PDAC_CD45pos_cac, PDAC_CD45pos_cac_dbl, PDAC_CD45pos_cac_filtered, PDAC_CD45pos_cac_norm, PDAC_CD45pos_cac_so, all_genes_PDAC_CD45neg_CTL, all_genes_PDAC_CD45pos_CTL, all_genes_PDAC_CD45neg_WS, all_genes_PDAC_CD45pos_WS, all_genes_PDAC_CD45neg_cac, all_genes_PDAC_CD45pos_cac)
```


## Merge
```{r, include=FALSE}
DefaultAssay(PDAC_CD45neg_CTL_singlets) <- "RNA"
DefaultAssay(PDAC_CD45pos_CTL_singlets) <- "RNA"
DefaultAssay(PDAC_CD45neg_WS_singlets) <- "RNA"
DefaultAssay(PDAC_CD45pos_WS_singlets) <- "RNA"
DefaultAssay(PDAC_CD45neg_cac_singlets) <- "RNA"
DefaultAssay(PDAC_CD45pos_cac_singlets) <- "RNA"


PDAC_combined <- merge(PDAC_CD45neg_CTL_singlets, y = list(PDAC_CD45pos_CTL_singlets, PDAC_CD45neg_WS_singlets, PDAC_CD45pos_WS_singlets, PDAC_CD45neg_cac_singlets, PDAC_CD45pos_cac_singlets), add.cell.ids = c("PDAC_CD45neg_CTL", "PDAC_CD45pos_CTL", "PDAC_CD45neg_WS", "PDAC_CD45pos_WS", "PDAC_CD45neg_cac", "PDAC_CD45pos_cac"))

PDAC_combined$orig.ident <- factor(PDAC_combined$orig.ident, levels = c("PDAC_CD45neg_CTL", "PDAC_CD45pos_CTL", "PDAC_CD45neg_WS", "PDAC_CD45pos_WS", "PDAC_CD45neg_cac", "PDAC_CD45pos_cac"))
```


## Pre-process

```{r, include=FALSE}
PDAC_combined <- NormalizeData(PDAC_combined) %>%
  FindVariableFeatures() %>%
  ScaleData() %>%
  RunPCA(npcs = 50)
```

```{r}
ElbowPlot(PDAC_combined, ndims = 50)
```


## Run harmony

```{r, include=FALSE}
PDAC_combined <- RunHarmony(PDAC_combined, group.by.vars = "orig.ident")

PDAC_combined <- FindNeighbors(PDAC_combined,reduction = "harmony", dims = 1:43) %>% 
  FindClusters(resolution = 0.1)

PDAC_combined <- RunUMAP(PDAC_combined, reduction = "harmony", dims = 1:43)
```


## Viualize data integration

```{r}
DimPlot(PDAC_combined, reduction = "harmony", group.by = "orig.ident", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "harmony1", y = "harmony2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```


## Remove unused objects from environment to improve processing

```{r, include=FALSE}
rm(PDAC_CD45neg_CTL_singlets, PDAC_CD45pos_CTL_singlets, PDAC_CD45neg_WS_singlets, PDAC_CD45pos_WS_singlets, PDAC_CD45neg_cac_singlets, PDAC_CD45pos_cac_singlets)
```



# 6. Annotate clusters

```{r}
DefaultAssay(PDAC_combined) <- "RNA"
Idents(PDAC_combined) <- "seurat_clusters"

cluster_ids <- levels(PDAC_combined$seurat_clusters)

DotPlot(object = PDAC_combined, features = c("PAX7", "MYF5", "ACTA1", "CDH5", "MYL9", "PDGFRB", "PDGFRA", "MMRN1", "CPA3", "CD68", "CSF1R", "S100A9", "NKG7", "IL7R", "HBA2")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, face = "italic")) +
  scale_color_gradientn(colors = c("red", "white", "blue")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, face = "italic"))
```


```{r, include=FALSE}
cluster_labels <- c("FAPs", 
                 "Myonuclei", 
                 "Neutrophils", 
                 "Endothelial cells",
                 "vSMCs/pericytes",
                 "Monocytes/macrophages",
                 "T cells/NK cells",
                 "Erythrocytes",
                 "MuSCs",
                 "Mast cells",
                 "Platelets",
                 "Myonuclei")

cluster_to_cell_type <- setNames(cluster_labels, cluster_ids)

PDAC_combined$seurat_clusters <- cluster_labels[PDAC_combined$seurat_clusters]

PDAC_combined <- JoinLayers(PDAC_combined, what = "all")
```


## Order clusters

```{r, include=FALSE}
PDAC_combined$seurat_clusters <- factor(PDAC_combined$seurat_clusters, levels = c("MuSCs", 
    "Myonuclei",
    "Endothelial cells",
    "vSMCs/pericytes",
    "FAPs",
    "Platelets",
    "Mast cells",
    "Monocytes/macrophages",
    "Neutrophils",
    "T cells/NK cells",
    "Erythrocytes"))
```


## Find markers between clusters

```{r, include=FALSE}
DEGs_PDAC_clusters <- FindAllMarkers(object = PDAC_combined, group.by = "seurat_clusters", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r, include=FALSE}
write.csv(DEGs_PDAC_clusters, file = "DEGs_PDAC_clusters.csv", row.names = FALSE)
```


## Remove Erythrocytes

```{r, include=FALSE}
PDAC_erythrocytes_removed <- "Erythrocytes"
PDAC_combined <- subset(PDAC_combined, seurat_clusters != PDAC_erythrocytes_removed)
```



# 7. Combine conditions

```{r, include=FALSE}
PDAC_combined$orig.ident <- ifelse(
  PDAC_combined$orig.ident %in% c("PDAC_CD45neg_CTL", "PDAC_CD45pos_CTL"), "PDAC_CTL",
  ifelse(PDAC_combined$orig.ident %in% c("PDAC_CD45neg_WS", "PDAC_CD45pos_WS"), "PDAC_WS", ifelse(PDAC_combined$orig.ident %in% c("PDAC_CD45neg_cac", "PDAC_CD45pos_cac"), "PDAC_cachectic", PDAC_combined$orig.ident)))
```

```{r, include=FALSE}
PDAC_combined$orig.ident <- factor(PDAC_combined$orig.ident, levels = c("PDAC_CTL", "PDAC_WS", "PDAC_cachectic"))
```



# 8. Visualize

## UMAP clusters

```{r}
DimPlot(PDAC_combined, reduction = "umap", group.by = "seurat_clusters", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```

## UMAP conditions

```{r}
DimPlot(PDAC_combined, reduction = "umap", group.by = "orig.ident", repel = TRUE, pt.size = 1) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank())
```


## Highlighted by condition

```{r}
DimPlot(
  PDAC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("PDAC_CTL" = WhichCells(PDAC_combined, expression = orig.ident == "PDAC_CTL")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#E87D72"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  PDAC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("PDAC_WS" = WhichCells(PDAC_combined, expression = orig.ident == "PDAC_WS")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#87AD34"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")

DimPlot(
  PDAC_combined,
  reduction = "umap",
  group.by = "seurat_clusters",
  cells.highlight = list("PDAC_cachectic" = WhichCells(PDAC_combined, expression = orig.ident == "PDAC_cachectic")),
  sizes.highlight = 1,
  pt.size = 1,
  cols.highlight = "#56BCC2"
) +
  ggplot2::labs(title = "", x = "UMAP1", y = "UMAP2") +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank()) +
  theme(legend.position = "none")
```


## Cell numbers per cluster for each time point

```{r}
PDAC_cell_counts_cluster <- table(PDAC_combined@meta.data$seurat_clusters, PDAC_combined@meta.data$orig.ident)

print(PDAC_cell_counts_cluster)
```



# 9. Subset


## Conditions

```{r, include=FALSE}
Idents(PDAC_combined) <- "orig.ident"

PDAC_CTL <- subset(PDAC_combined, idents = "PDAC_CTL")
PDAC_WS <- subset(PDAC_combined, idents = "PDAC_WS")
PDAC_cac <- subset(PDAC_combined, idents = "PDAC_cachectic")
```


## Clusters

```{r, include=FALSE}
Idents(PDAC_combined) <- "seurat_clusters"

PDAC_MuSCs <- subset(PDAC_combined, idents = "MuSCs")
PDAC_Myonuclei <- subset(PDAC_combined, idents = "Myonuclei")
PDAC_Endothelial_cells <- subset(PDAC_combined, idents = "Endothelial cells")
PDAC_vSMCs_pericytes <- subset(PDAC_combined, idents = "vSMCs/pericytes")
PDAC_FAPs <- subset(PDAC_combined, idents = "FAPs")
PDAC_Platelets <- subset(PDAC_combined, idents = "Platelets")
PDAC_Mast_cells <- subset(PDAC_combined, idents = "Mast cells")
PDAC_Monocytes_macrophages <- subset(PDAC_combined, idents = "Monocytes/macrophages")
PDAC_Neutrophils <- subset(PDAC_combined, idents = "Neutrophils")
PDAC_T_NK_cells <- subset(PDAC_combined, idents = "T cells/NK cells")
```


## DEGs per condition for each cluster

```{r, include=FALSE}
Idents(PDAC_MuSCs) <- "orig.ident"
Idents(PDAC_Myonuclei) <- "orig.ident"
Idents(PDAC_Endothelial_cells) <- "orig.ident"
Idents(PDAC_vSMCs_pericytes) <- "orig.ident"
Idents(PDAC_FAPs) <- "orig.ident"
Idents(PDAC_Platelets) <- "orig.ident"
Idents(PDAC_Mast_cells) <- "orig.ident"
Idents(PDAC_Monocytes_macrophages) <- "orig.ident"
Idents(PDAC_Neutrophils) <- "orig.ident"
Idents(PDAC_T_NK_cells) <- "orig.ident"

DEGs_PDAC_MuSCs <- FindAllMarkers(object = PDAC_MuSCs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Myonuclei <- FindAllMarkers(object = PDAC_Myonuclei, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Endothelial_cells <- FindAllMarkers(object = PDAC_Endothelial_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_vSMCs_pericytes <- FindAllMarkers(object = PDAC_vSMCs_pericytes, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_FAPs <- FindAllMarkers(object = PDAC_FAPs, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Platelets <- FindAllMarkers(object = PDAC_Platelets, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Mast_cells <- FindAllMarkers(object = PDAC_Mast_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Monocytes_macrophages <- FindAllMarkers(object = PDAC_Monocytes_macrophages, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_Neutrophils <- FindAllMarkers(object = PDAC_Neutrophils, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
DEGs_PDAC_T_NK_cells <- FindAllMarkers(object = PDAC_T_NK_cells, group.by = "orig.ident", test.use = "LR", only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```



# Visualize PDAC MuSC gene expression (from mouse cachectic subcluster list)

```{r}
common_cachectic_genes <- c(
 "FTH1", "LMNA", "EIF1", "PNRC1", "NFKBIA", "CEBPB", "IFITM3", "SAT1" 
)

Idents(PDAC_MuSCs) <- "orig.ident"

vln_plots_mouse_genes <- lapply(common_cachectic_genes, function(gene) {
  VlnPlot(PDAC_MuSCs, features = gene, pt.size = 0) +
    theme(
      legend.position = "none",
      axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)
    ) +
    labs(x = NULL, y = "Expression", title = gene)
})

do.call(grid.arrange, c(vln_plots_mouse_genes, ncol = 4))
```


## Expression values

```{r}
common_cachectic_genes_expression <- FetchData(PDAC_MuSCs, vars = c(common_cachectic_genes, "orig.ident"))

common_cachectic_genes_expression <- common_cachectic_genes_expression %>%
  group_by(orig.ident) %>%
  summarise(across(all_of(common_cachectic_genes), mean, .names = "avg_{.col}"))

common_cachectic_genes_expression <- common_cachectic_genes_expression %>%
  column_to_rownames("orig.ident") %>%
  t() %>%
  as.data.frame()

print(common_cachectic_genes_expression)
```



# Visualize PDAC MuSC gene expression (from PDAC DEGs list)

```{r}
top_cachectic_genes <- c(
 "MT-CO2", "TRDN", "MT-ND4", "TPT1", "MT-ND2", "MT-CO1", "DES", "TNNC2", "RPL7" 
)

Idents(PDAC_MuSCs) <- "orig.ident"

vln_plots_PDAC <- lapply(top_cachectic_genes, function(gene) {
  VlnPlot(PDAC_MuSCs, features = gene, pt.size = 0) +
    theme(
      legend.position = "none",
      axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)
    ) +
    labs(x = NULL, y = "Expression", title = gene)
})

do.call(grid.arrange, c(vln_plots_PDAC, ncol = 4))

FetchData(PDAC_MuSCs, vars = c(top_cachectic_genes, "orig.ident"))
```


## Expression values

```{r}
top_cachectic_genes_expression <- FetchData(PDAC_MuSCs, vars = c(top_cachectic_genes, "orig.ident"))

top_cachectic_genes_expression <- top_cachectic_genes_expression %>%
  group_by(orig.ident) %>%
  summarise(across(all_of(top_cachectic_genes), mean, .names = "avg_{.col}"))

top_cachectic_genes_expression <- top_cachectic_genes_expression %>%
  column_to_rownames("orig.ident") %>%
  t() %>%
  as.data.frame()

print(top_cachectic_genes_expression)
```
